clients/src/main/java/org/apache/kafka/common/security/token/delegation/internals/DelegationTokenCache.java

```diff
@@ -32,10 +32,15 @@
 public class DelegationTokenCache {
 
     private CredentialCache credentialCache = new CredentialCache();
+
     //Cache to hold all the tokens
     private Map<String, TokenInformation> tokenCache = new ConcurrentHashMap<>();
+
     //Cache to hold hmac->tokenId mapping. This is required for renew, expire requests
-    private Map<String, String> hmacIDCache = new ConcurrentHashMap<>();
+    private Map<String, String> hmacTokenIdCache = new ConcurrentHashMap<>();
+
+    //Cache to hold tokenId->hmac mapping. This is required for removing entry from hmacTokenIdCache using tokenId.
+    private Map<String, String> tokenIdHmacCache = new ConcurrentHashMap<>();
```

Comment:
Added this instance instead of `hmacTokenIdCache` as BiMap as it requires google guava dependency and not much of that API is needed here.